crates/worker/src/partition/state_machine/command_interpreter.rs

```diff
@@ -515,6 +641,91 @@ where
         Ok(())
     }
 
+    async fn cancel_journal_leaves<State: StateReader>(
+        &mut self,
+        full_invocation_id: FullInvocationId,
+        invocation_status: InvocationStatusProjection,
+        journal_length: EntryIndex,
+        state: &mut State,
+        effects: &mut Effects,
+    ) -> Result<bool, Error> {
+        let mut journal = state.get_journal(&full_invocation_id.service_id, journal_length);
+
+        let canceled_result = CompletionResult::Failure(
+            UserErrorCode::from(CANCELED_INVOCATION_ERROR.code()),
+            CANCELED_INVOCATION_ERROR.message().into(),
+        );
+
+        let mut resume_invocation = false;
+
+        while let Some(journal_entry) = journal.next().await {
+            let (journal_index, journal_entry) = journal_entry?;
+
+            if let JournalEntry::Entry(journal_entry) = journal_entry {
+                match journal_entry.header {
+                    // cancel uncompleted invocations
+                    EnrichedEntryHeader::Invoke {
+                        is_completed,
+                        resolution_result: Some(resolution_result),
+                    } if !is_completed => {
+                        let target_fid = FullInvocationId::new(
+                            resolution_result.service_name,
+                            resolution_result.service_key,
+                            resolution_result.invocation_uuid,
+                        );
+
+                        self.send_message(
+                            OutboxMessage::TerminateInvocation(InvocationTermination::cancel(
+                                target_fid,
+                            )),
+                            effects,
+                        );
+                    }
+                    EnrichedEntryHeader::Awakeable { is_completed }
+                    | EnrichedEntryHeader::GetState { is_completed }
```

Comment:
You can't do this, because in the protocol `GetState` doesn't have a failure variant.